### PR TITLE
Authority Migration Cache read changes

### PR DESCRIPTION
**Pull request by RPangrle**
* Use the aliases from the Authority validation cache to look up tokens in the token cache accessor

Note, this change is only for when we’re reading tokens to attempt to use them on the network, not when doing sanity checks or writing out afterwards. Changes to the writing behavior will be a different PR.

### CODE REVIEW

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 1]
```diff
@@ -63,25 +64,45 @@ - (id)initWithDataSource:(id<ADTokenCacheDataSource>)dataSource
...
+    for (NSURL *alias in aliases)
     {
-        return nil;
+        ADTokenCacheKey* key = [ADTokenCacheKey keyWithAuthority:[alias absoluteString]
+                                                        resource:resource
+                                                        clientId:clientId
+                                                           error:error];
+        if (!key)
+        {
+            return nil;
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: Just theoretically: if we fail to create a key with one given alias, maybe we should try with the next one instead of returning early?

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 2]
```diff
@@ -63,25 +64,45 @@ - (id)initWithDataSource:(id<ADTokenCacheDataSource>)dataSource
...
+            return item;
+        }
+        
+        if (adError)
+        {
+            if (error)
+            {
+                *error = adError;
+            }
+            return nil;
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: nit: it will return nil in the end anyway

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 3]
```diff
@@ -63,25 +64,45 @@ - (id)initWithDataSource:(id<ADTokenCacheDataSource>)dataSource
     return _dataSource;
 }
 
-- (ADTokenCacheItem *)getItemForUser:(ADUserIdentifier *)identifier
+- (ADTokenCacheItem *)getItemForUser:(NSString *)userId
```
Comment (oldalton) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: just curious: why was this change necessary?

File: ADAL/src/cache/ADTokenCacheAccessor.m (hunk 1 above)
Comment (RPangrle) [reply ADAL/src/cache/ADTokenCacheAccessor.m]: If we fail to return a key then we have some bigger problem that I'd prefer not to mask seeing how the key is little more then a bunch of strings glued together.

File: ADAL/src/cache/ADTokenCacheAccessor.m (hunk 3 above)
Comment (RPangrle) [reply ADAL/src/cache/ADTokenCacheAccessor.m]: It was a change I made because on the write side we have a userId not an identifier. This PR is from changes I pulled out to focus on read.
